cuML/src/umap/simpl_set_embed/algo.h

```diff
@@ -105,9 +110,9 @@ namespace UMAPAlgo {
 	         */
 	        template<typename T>
 	        __device__ __host__ T attractive_grad(T dist_squared, UMAPParams params) {
-                T grad_coeff = -2.0 * params.a * params.b *
-                        pow(dist_squared, params.b - 1.0);
-                grad_coeff /= params.a * pow(dist_squared, params.b) + 1.0;
+                double grad_coeff = -2.0 * double(params.a) * double(params.b) *
```

Comment:
Out of curiosity, why `double` for everything now? Is that slow for device code?

Reply:
It's great you point this out, actually. This was an artifact left behind from debugging. 

Ideally, all of our algorithms would support both single and double precision inputs. We are only able to support single precision in this algorithm for now because FAISS' KNN only supports supports single precision.